File: buzzvil-sdk-ios-v6/buzzvil-sdk-ios-objc/ViewController.m
#import "ViewController.h"
#import "UIButton+Custom.h"
#import "Native/NativeViewController.h"
#import "Native/Carousel/CarouselViewController.h"
#import "Interstitial/InterstitialViewController.h"
#import "BenefitHub/BenefitHubViewController.h"
#import "BenefitHub/BenefitHubContainerViewController.h"
#import "Banner/BannerViewController.h"

@interface ViewController ()

@property (nonatomic, strong, readonly) UIStackView *rootStackView;
@property (nonatomic, strong, readonly) UIButton *nativeButton;
@property (nonatomic, strong, readonly) UIButton *carouselButton;
@property (nonatomic, strong, readonly) UIButton *interstitialButton;
@property (nonatomic, strong, readonly) UIButton *benefitHubButton;
@property (nonatomic, strong, readonly) UIButton *benefitHubContainerButton;
@property (nonatomic, strong, readonly) UIButton *bannerButton;

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  [self setupView];
  [self setupLayout];
  [self setupEvent];
}

#pragma mark - UI setup
- (void)setupView {
  self.view.backgroundColor = [UIColor systemBackgroundColor];
  self.navigationItem.title = @"BuzzvilSDK-ObjectiveC";
  
  _rootStackView = [[UIStackView alloc] initWithFrame:CGRectZero];
  [self.view addSubview:_rootStackView];
  _rootStackView.axis = UILayoutConstraintAxisVertical;
  _rootStackView.spacing = 12;
  _rootStackView.distribution = UIStackViewDistributionFillEqually;
  
  _nativeButton = [UIButton buttonWithType:UIButtonTypeSystem];
  [_nativeButton setTitle:@"Native" forState:UIControlStateNormal];
  [_nativeButton applyCustomStyle];
  
  _interstitialButton = [UIButton buttonWithType:UIButtonTypeSystem];
  [_interstitialButton setTitle:@"Interstitial" forState:UIControlStateNormal];
  [_interstitialButton applyCustomStyle];
  
  _carouselButton = [UIButton buttonWithType:UIButtonTypeSystem];
  [_carouselButton setTitle:@"Carousel" forState:UIControlStateNormal];
  [_carouselButton applyCustomStyle];
  
  _benefitHubButton = [UIButton buttonWithType:UIButtonTypeSystem];
  [_benefitHubButton setTitle:@"BenefitHub" forState:UIControlStateNormal];
  [_benefitHubButton applyCustomStyle];
  
  _benefitHubContainerButton = [UIButton buttonWithType:UIButtonTypeSystem];
  [_benefitHubContainerButton setTitle:@"BenefitHub Container" forState:UIControlStateNormal];
  [_benefitHubContainerButton applyCustomStyle];
  
  _bannerButton = [UIButton buttonWithType:UIButtonTypeSystem];
  [_bannerButton setTitle:@"Banner" forState:UIControlStateNormal];
  [_bannerButton applyCustomStyle];
  
  [_rootStackView addArrangedSubview: _nativeButton];
  [_rootStackView addArrangedSubview: _interstitialButton];
  [_rootStackView addArrangedSubview: _benefitHubButton];
  [_rootStackView addArrangedSubview: _benefitHubContainerButton];
  [_rootStackView addArrangedSubview: _bannerButton];
}

- (void) setupLayout {
  _rootStackView.translatesAutoresizingMaskIntoConstraints = NO;
  [NSLayoutConstraint activateConstraints:@[
    [_rootStackView.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor],
    [_rootStackView.leadingAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.leadingAnchor],
    [_rootStackView.trailingAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.trailingAnchor],
    [_rootStackView.bottomAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.bottomAnchor],
  ]];
}


- (void)setupEvent {
  [_benefitHubButton addGestureRecognizer:[[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(pushBenefitHubViewController:)]];
  [_nativeButton addGestureRecognizer:[[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(pushNativeViewController:)]];
  [_carouselButton addGestureRecognizer:[[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(pushCarouselViewController:)]];
  [_interstitialButton addGestureRecognizer:[[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(pushInterstitialViewController:)]];
  [_benefitHubContainerButton addGestureRecognizer:[[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(pushBenefitHubContainerViewController:)]];
  [_bannerButton addGestureRecognizer:[[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(pushBannerViewController:)]];
}

- (void)pushBenefitHubViewController:(id)sender {
  BenefitHubViewController *feedViewController = [[BenefitHubViewController alloc] init];
  [self.navigationController pushViewController:feedViewController animated:YES];
}

- (void)pushNativeViewController:(id)sender {
  NativeViewController *nativeViewController = [[NativeViewController alloc] init];
  [self.navigationController pushViewController:nativeViewController animated:YES];
}

- (void)pushCarouselViewController:(id)sender {
  CarouselViewController *carouselViewController = [[CarouselViewController alloc] init];
  [self.navigationController pushViewController:carouselViewController animated:YES];
}

- (void)pushInterstitialViewController:(id)sender {
  InterstitialViewController *interstitialViewController = [[InterstitialViewController alloc] init];
  [self.navigationController pushViewController:interstitialViewController animated:YES];
}

- (void)pushBenefitHubContainerViewController:(id)sender {
  BenefitHubContainerViewController *feedContainerViewController = [[BenefitHubContainerViewController alloc] init];
  [self.navigationController pushViewController:feedContainerViewController animated:YES];
}

- (void)pushBannerViewController:(id)sender {
  BannerViewController *bannerViewController = [[BannerViewController alloc] init];
  [self.navigationController pushViewController:bannerViewController animated:YES];
}

@end

File: buzzvil-sdk-ios-v6/buzzvil-sdk-ios-swift/ViewController.swift
import UIKit
import BuzzvilSDK

class ViewController: UIViewController {
  private static let navigationItemTitle = "BuzzvilSDK"
  private static let nativeButtonTitle = "Native"
  private static let interstitialButtonTitle = "Interstitial"
  private static let buzzAdBenefitHubButtonTitle = "BenefitHub"
  private static let buzzAdBenefitHubConatinerButtonTitle = "BenefitHub Container"
  private static let buzzAdBannerButtonTitle = "Banner"
  private static let buzzAdMissionPackButtonTitle = "MissionPack"
  private static let carouselButtonTitle = "Carousel"
  private static let inquiryButtonTitle = "Inquiry"
  private static let buttonAspectRatio: CGFloat = 1.5
  private static let buttonMargin: CGFloat = 12
  private static let buttonCornerRadius: CGFloat = 8
  private static let buttonFontSize: CGFloat = 16
  
  private lazy var rootStackView: UIStackView = {
    let stackView = UIStackView(frame: .zero)
    stackView.axis = .vertical
    stackView.spacing = Self.buttonMargin
    stackView.distribution = .fillEqually
    return stackView
  }()
  
  private lazy var nativeButton: UIButton = {
    let button = UIButton(frame: .zero)
    button.setTitle(Self.nativeButtonTitle, for: .normal)
    button.addTarget(self, action: #selector(pushNativeAdViewController), for: .touchUpInside)
    return button
  }()
  
  private lazy var interstitialButton: UIButton = {
    let button = UIButton(frame: .zero)
    button.setTitle(Self.interstitialButtonTitle, for: .normal)
    button.addTarget(self, action: #selector(pushInterstitialViewController), for: .touchUpInside)
    return button
  }()
  
  private lazy var carouselButton: UIButton = {
    let button = UIButton(frame: .zero)
    button.setTitle(Self.carouselButtonTitle, for: .normal)
    button.addTarget(self, action: #selector(pushCarouselViewController), for: .touchUpInside)
    return button
  }()
  
  private lazy var benefitHubButton: UIButton = {
    let button = UIButton(frame: .zero)
    button.setTitle(Self.buzzAdBenefitHubButtonTitle, for: .normal)
    button.addTarget(self, action: #selector(pushFeedViewController), for: .touchUpInside)
    return button
  }()
  
  private lazy var benefitHubContainerButton: UIButton = {
    let button = UIButton(frame: .zero)
    button.setTitle(Self.buzzAdBenefitHubConatinerButtonTitle, for: .normal)
    button.addTarget(self, action: #selector(pushFeedContainerViewController), for: .touchUpInside)
    return button
  }()
  
  private lazy var bannerButton: UIButton = {
    let button = UIButton(frame: .zero)
    button.setTitle(Self.buzzAdBannerButtonTitle, for: .normal)
    button.addTarget(self, action: #selector(pushBannerViewController), for: .touchUpInside)
    return button
  }()
  
  private lazy var inquiryButton: UIButton = {
    let button = UIButton(frame: .zero)
    button.setTitle(Self.inquiryButtonTitle, for: .normal)
    button.addTarget(self, action: #selector(showInquiry), for: .touchUpInside)
    return button
  }()
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
    setupView()
    setupLayout()
  }
  
  private func setupView() {
    view.backgroundColor = .systemBackground

    navigationItem.title = Self.navigationItemTitle
    
    rootStackView.addArrangedSubview(nativeButton)
    rootStackView.addArrangedSubview(interstitialButton)
    rootStackView.addArrangedSubview(carouselButton)
    rootStackView.addArrangedSubview(benefitHubButton)
    rootStackView.addArrangedSubview(benefitHubContainerButton)
    rootStackView.addArrangedSubview(bannerButton)
    
    view.addSubview(rootStackView)
    
    let buttonsToSetupAppearance = [
      nativeButton,
      interstitialButton,
      carouselButton,
      benefitHubButton,
      benefitHubContainerButton,
      bannerButton,
    ]
    
    buttonsToSetupAppearance.forEach { button in
      setupAppearanceOfButton(button: button)
    }
  }
  
  private func setupLayout() {
    rootStackView.translatesAutoresizingMaskIntoConstraints = false
    NSLayoutConstraint.activate([
      rootStackView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
      rootStackView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
      rootStackView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
      rootStackView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
    ])
  }
  
  private func setupAppearanceOfButton(button: UIButton) {
    button.backgroundColor = .systemBlue
    button.layer.cornerRadius = Self.buttonCornerRadius
    button.setTitleColor(.white, for: .normal)
    button.titleLabel?.font = UIFont.systemFont(ofSize: Self.buttonFontSize)
    button.translatesAutoresizingMaskIntoConstraints = false
    NSLayoutConstraint.activate([
      button.heightAnchor.constraint(equalToConstant: 32)
    ])
  }
  
  @objc private func pushNativeAdViewController() {
    let nativeAdViewController = NativeViewController()
    navigationController?.pushViewController(nativeAdViewController, animated: true)
  }
  
  @objc private func pushCarouselViewController() {
    let carouselConfigurationViewController = CarouselViewController()
    navigationController?.pushViewController(carouselConfigurationViewController, animated: true)
  }
  
  @objc private func pushInterstitialViewController() {
    let interstitialViewController = InterstitialViewController()
    navigationController?.pushViewController(interstitialViewController, animated: true)
  }
  
  @objc private func pushFeedViewController() {
    let feedViewController = BenefitHubViewController()
    navigationController?.pushViewController(feedViewController, animated: true)
  }
  
  @objc private func pushFeedContainerViewController() {
    let feedContainerViewController = BenefitHubContainerViewController()
    navigationController?.pushViewController(feedContainerViewController, animated: true)
  }
  
  @objc private func pushBannerViewController() {
    let bannerViewController = BannerViewController()
    navigationController?.pushViewController(bannerViewController, animated: true)
  }
  
  @objc private func showInquiry() {
    BuzzAdBenefit.presentInquiryPage(on: self, unitId: "YOUR_UNIT_ID")
  }
}
